### PR TITLE
Module 7 nodejs

### PR DESCRIPTION
**Pull request by danemesis**

### CODE REVIEW

File: Module7_NodeJS/.idea/misc.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="JavaScriptSettings">
+    <option name="languageLevel" value="JSX" />
+  </component>
+</project>
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/.idea/misc.xml:6]: it is better move .idea folder to gitignore

File: Module7_NodeJS/node-workshop/frontend/src/api/devices.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+const getDevices = () => fetch('/api/device')
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/node-workshop/frontend/src/api/devices.js]: please use variables for endpoint value

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/Group.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    groupNameChange(e) {
+        this.setState({
+            groupName: e.target.value
+        });
+    }
+
+    onSubmit(e) {
+        e.preventDefault();
+
+        fetch('/api/group', {
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/node-workshop/frontend/src/app/Group/Group.js]: please move 'api/group' to variable

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                        onClick={this.changeDeviceGroup.bind(this, device.name, group)}
+                        key={index}>
+                    {device.name}
+                </button>
+            )
+        }
+    }
+
+    changeDeviceGroup(deviceName, group) {
+        fetch(`/api/group/device/`, {
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js]: here too

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            method: 'POST',
+            headers: {
+                Accept: 'application/json',
+                'Content-Type': 'application/json'
+            },
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js]: can we set headers for all requests just once? like interceptor in angular?

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            if (array[i] === checker) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    changeAllDevicesInGroup(group, action) {
+        fetch(`/api/group/${action}/`, {
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js]: see above

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+            method: 'POST',
+            headers: {
+                Accept: 'application/json',
+                'Content-Type': 'application/json'
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js]: see above

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/List/List.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+        this.state = {
+            data: {}
+        };
+
+        this.fetchList = this.fetchList.bind(this);
+    }
+
+    fetchList() {
+        fetch('/api/group', {
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/node-workshop/frontend/src/app/Group/List/List.js]: see above

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/List/List.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+            method: 'GET',
+            headers: {
+                Accept: 'application/json',
+                'Content-Type': 'application/json'
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/node-workshop/frontend/src/app/Group/List/List.js]: see above

File: Module7_NodeJS/node-workshop/frontend/src/app/Main/Main.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+        this.state = {devices: []};
+
+        this.updateDeviceList = this.updateDeviceList.bind(this);
+        this.delDevice = this.delDevice.bind(this);
+        this.addDevice = this.addDevice.bind(this);
+    }
+
+    updateDeviceList() {
+        fetch('/api/device')
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/node-workshop/frontend/src/app/Main/Main.js]: we have method getDevices in api/devices.js, please use it or remove

File: Module7_NodeJS/node-workshop/server/controllers/group.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+        name: new_group.name,
+        devices: new_group.devices
+    });
+
+    logging(createdGroup.name, 'device group was created.');
+
+    res.sendStatus(200);
+});
+
+router.delete('/:name', async (req, res) => {
```
Comment (YaroslavPerepelychnyi) [inline Module7_NodeJS/node-workshop/server/controllers/group.js]: did you implement delete?

File: Module7_NodeJS/.idea/misc.xml (hunk 1 above)
Comment (danemesis) [reply Module7_NodeJS/.idea/misc.xml:6]: moved

File: Module7_NodeJS/node-workshop/server/controllers/group.js (hunk 11 above)
Comment (danemesis) [reply Module7_NodeJS/node-workshop/server/controllers/group.js]: no :) deleted

File: Module7_NodeJS/node-workshop/frontend/src/app/Main/Main.js (hunk 10 above)
Comment (danemesis) [reply Module7_NodeJS/node-workshop/frontend/src/app/Main/Main.js]: deleted

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/List/List.js (hunk 8 above)
Comment (danemesis) [reply Module7_NodeJS/node-workshop/frontend/src/app/Group/List/List.js]: fixed

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js (hunk 6 above)
Comment (danemesis) [reply Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js]: fixed

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js (hunk 4 above)
Comment (danemesis) [reply Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js]: fixed

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/Group.js (hunk 3 above)
Comment (danemesis) [reply Module7_NodeJS/node-workshop/frontend/src/app/Group/Group.js]: fixed

File: Module7_NodeJS/node-workshop/frontend/src/api/devices.js (hunk 2 above)
Comment (danemesis) [reply Module7_NodeJS/node-workshop/frontend/src/api/devices.js]: deleted

File: Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js (hunk 5 above)
Comment (danemesis) [reply Module7_NodeJS/node-workshop/frontend/src/app/Group/GroupsElement/GroupsElement.js]: done 
at api/api.js :)

**Review by YaroslavPerepelychnyi — APPROVED** [review]

**Pull request merged**